File: cs/Minsk/CodeAnalysis/Binding/BoundVariableDeclaration.cs
namespace Minsk.CodeAnalysis.Binding;

internal sealed class BoundVariableDeclaration : BoundStatement
{
    public VariableSymbol Variable { get; }
    public BoundExpression Initializer { get; }

    public BoundVariableDeclaration(VariableSymbol variable, BoundExpression initializer)
    {
        Variable = variable;
        Initializer = initializer;
    }

    public override BoundNodeKind Kind => BoundNodeKind.VariableDeclaration;
}

File: kt/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/binding/BoundVariableDeclaration.kt
package dev.phytolizer.minsk.analysis.binding

import dev.phytolizer.minsk.analysis.VariableSymbol

internal class BoundVariableDeclaration(val variable: VariableSymbol, val initializer: BoundExpression) :
    BoundStatement() {
    override val kind: BoundNodeKind
        get() = BoundNodeKind.VariableDeclaration
}
